contracts/registry/src/test.rs

```diff
@@ -129,21 +129,22 @@ fn validate_version() {
     let (client, address) = &init();
     let env = env();
     let name = &to_string("registry");
-    let bytes = Bytes::from_slice(env, registry::WASM);
+    let bytes = &Bytes::from_slice(env, registry::WASM);
     env.mock_all_auths();
-    let version = default_version();
-    client.publish(name, address, &bytes, &version);
+    let version = &to_string("0.0.0");
+    let new_version = &to_string("0.0.1");
+    client.publish(name, address, bytes, version);
     assert_eq!(
-        client.try_publish(name, address, &bytes, &version),
+        client.try_publish(name, address, bytes, version),
         Err(Ok(Error::VersionMustBeGreaterThanCurrent))
     );
     assert_eq!(
-        client.try_publish(name, address, &bytes, &to_string("0.  0.0"),),
+        client.try_publish(name, address, bytes, &to_string("0.  0.0"),),
         Err(Ok(Error::InvalidVersion))
     );
-    client.publish(name, address, &bytes, &to_string("0.0.1"));
+    client.publish(name, address, bytes, new_version);
```

Comment:
Perhaps a silly question since this was already here -- but wouldn't we expect this publish to succeed? The new version is greater than current

Reply:
It does! I could make it clearer by using the `try_publish` version and checking `Ok(Ok(())` but in this case the function panics if the method fails. Then the next test tries to publish the old version and it fails.

Reply:
ohhh I see how I misread it, this is an unsaved value that publishes the next version, and the following assert is trying to publish older version. I get it.